app/models/auth_sources/auth_source_ldap.rb

```diff
@@ -130,6 +131,20 @@ def users_in_group(name)
     ldap_con.user_list(name)
   end
 
+  def test_connection
+    result = {}
+    begin
+      Timeout::timeout(60) do
```

Comment:
I'd suggest a slightly shorter time, perhaps 10-20 seconds?  I don't think it'll make any difference to a normal LDAP server, and it'll make the UI error show faster.